server/schemas/survey.js

```diff
@@ -44,17 +43,34 @@ export const typeDef = `
 export const resolvers = {
   Query: {
     survey: (root, { id }, { db }) => db.survey.findByPk(id),
-    surveys: (root, args, { db }) =>
-      db.survey.findAll({
-        order: [["createdAt", "ASC"]],
-      }),
+    surveys: (root, args, { db, currentUserId }) => {
+      if (currentUserId) {
+        return db.user.findByPk(currentUserId).then((user) =>
+          user.role === "ADMIN"
+            ? db.survey.findAll({
+                order: [["title", "ASC"]],
+              })
+            : db.survey.findAll({
+                where: { published: true },
+                order: [["title", "ASC"]],
+              })
+        );
+      } else {
+        return db.survey.findAll({
+          where: { published: true },
+          order: [["title", "ASC"]],
+        });
+      }
+    },
   },
   Mutation: {
     createSurvey: authenticated((root, { input }, { db, currentUserId }) =>
       db.survey
         .create({
           title: input.title,
+          opened: input.opened,
```

Comment:
[Note] I don't think people can create closed survies, that's why I removed the opened param from this create